components/file_system/src/thread_io.rs

```diff
@@ -0,0 +1,258 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::borrow::BorrowMut;
+use std::cell::Cell;
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+use std::path::PathBuf;
+use std::str::FromStr;
+use std::sync::{
+    atomic::{AtomicU64, Ordering},
+    Arc, Mutex, MutexGuard,
+};
+use std::time::Duration;
+
+use nix::unistd::{getpid, gettid, Pid};
```

Comment:
`gettid` is not available on `macOS`. So, this PR breaks our builds on macOS. Could you please switch it to `thread-id` to finish it?

Reply:
@MuZhou233 PTAL

Reply:
As I know, `thread-id` is actually `pthread id` on linux, Which is different from `tid` here. And `macOS` didn't have `/proc` for us to read.
I think it's better to mark this as linux only. But how do I deal with other target os? @Connor1996

Reply:
let's target it only for linux

Reply:
@MuZhou233 Will you try to fix it by limiting the platform? Or I can do it for you.

Reply:
@sticnarf I will open a new PR to fix it tonight.